public/app/features/playlist/api.ts

```diff
@@ -13,6 +13,10 @@ export async function updatePlaylist(id: number, playlist: Playlist) {
   await withErrorHandling(async () => await getBackendSrv().put(`/api/playlists/${id}`, playlist));
 }
 
+export async function deletePlaylist(id: number) {
+  await withErrorHandling(async () => await getBackendSrv().delete(`/api/playlists/${id}`));
+}
+
```

Comment:
Maybe use this suggestion:
```suggestion
export async function deletePlaylist(id: number) {
  await withErrorHandling(async () => await getBackendSrv().request({
  method: 'DELETE',
  url:`/api/playlists/${id}`,
  showSuccessAlert: false,
})
);
}

```

Reply:
Seems like the `request` function is deprecated?

Reply:
Maybe it's worth using `fetch` instead? @hugohaggmark

Reply:
Sure, go ahead. That uses Observables though, so you need to figure that out. 🤔

Reply:
Getting some errors with types
`Type 'Promise<FetchResponse<unknown>>' is not assignable to type 'Promise<void>'.`

maybe I'm doing this the wrong way?
```ts
await lastValueFrom(
    getBackendSrv().fetch({
      method: 'DELETE',
      url: `/api/playlists/${id}`,
      showSuccessAlert: false,
    })
  )
```

Reply:
> Sure, go ahead. That uses Observables though, so you need to figure that out. 

That's what I was trying to say you need to figure this out if you're using `fetch` instead of `request` 😉

Reply:
I guess we need to create a duplicate of the `withErrorHandling` function that has a type compatible with observables and return the correct alert message? @hugohaggmark 
```ts
async function withErrorHandling2(apiCall: () => Promise<FetchResponse<unknown>>) {
  try {
    await apiCall();
    dispatch(notifyApp(createSuccessNotification('Playlist deleted')));
  } catch (e) {
    dispatch(notifyApp(createErrorNotification('Unable to delete playlist', e)));
  }
}
```

Reply:
Nope, read the deprecation notice for `request` carefully again.

Reply:
oh...just found where the alert message is coming from!
We could as well use the `getBackendSrv().delete()` but wrap it up in `withErrorHandling2` function? I think I prefer this approach 😝